cdpctl/validation/infra/validate_azure_security_groups.py

```diff
@@ -0,0 +1,434 @@
+    AZURE_NSG_NOT_FOUND,
+    AZURE_SUBNETS_NOT_COVERED_BY_NSG,
+    AZURE_SUBNETS_NOT_VALID_NSG,
+    AZURE_VNET_NOT_FOUND,
+)
+
+_info = {}
+
+
+@pytest.fixture(autouse=True, name="azure_network_client")
+def azure_network_client_fixture(config: Dict[str, Any]) -> NetworkManagementClient:
+    """Return an Azure Client."""
+    return get_client("network", config)
+
+
+def _load_vnet_info(vnet_name, resource_group_name, azure_network_client):
+    if "vnet" not in _info:
+        try:
+            _info["vnet"] = azure_network_client.virtual_networks.get(
+                virtual_network_name=vnet_name, resource_group_name=resource_group_name
+            )
+        except ResourceNotFoundError:
+            fail(AZURE_VNET_NOT_FOUND, [vnet_name, resource_group_name])
+
+
+@pytest.fixture(name="azure_resource_group_name")
+def azure_resource_group_name_fixture(config: Dict[str, Any]):
+    """Return the resource group name."""
+    return get_config_value(config=config, key="infra:azure:metagroup:name")
+
+
+@pytest.fixture(name="azure_vnet_name")
+def azure_vnet_name_fixture(config: Dict[str, Any]):
+    """Return the vnet name."""
+    return get_config_value(config=config, key="infra:vpc:name")
+
+
+@pytest.fixture(name="azure_vnet_info")
+def azure_vnet_info_fixture(config, azure_network_client):
+    """Return the vnet info."""
+    vnet_name = get_config_value(config=config, key="infra:vpc:name")
+    resource_group_name = get_config_value(
+        config=config, key="infra:azure:metagroup:name"
+    )
+    _load_vnet_info(
+        vnet_name=vnet_name,
+        resource_group_name=resource_group_name,
+        azure_network_client=azure_network_client,
+    )
+    return _info["vnet"]
+
+
+@pytest.fixture(name="azure_default_nsg_info")
+def azure_default_nsg_info_fixture():
+    """Return the default nsg info."""
+    return _info["default_nsg"]
+
+
+@pytest.fixture(name="azure_knox_nsg_info")
+def azure_knox_nsg_info_fixture():
+    """Return the knox NSG info."""
+    return _info["knox_nsg"]
+
+
+@pytest.mark.azure
+@pytest.mark.infra
+@pytest.mark.dependency(
+    depends=[
+        "infra/validate_azure_region.py::azure_region_validation",
+    ],
+    scope="session",
+)
+def azure_default_security_group_exists_validation(
+    config: Dict[str, Any],
+    azure_network_client: NetworkManagementClient,
+) -> None:  # pragma: no cover
+    """Default Security Group exists."""  # noqa: D401
+    azure_default_security_group_exists(config, azure_network_client)
+
+
+@validator
+def azure_default_security_group_exists(
+    config: Dict[str, Any],
+    azure_network_client: NetworkManagementClient,
+) -> None:  # pragma: no cover
+    """Check that the Security Group exists."""  # noqa: D401,E501
+    resource_group_name = get_config_value(
+        config=config, key="infra:azure:metagroup:name"
+    )
+    nsg_name = get_config_value(config=config, key="env:security_group:default:name")
```

Comment:
I see it under `infra:security_group`

https://github.com/cloudera-labs/cloudera.exe/blob/main/docs/configuration.yml#L287

Reply:
yeap my bad